func_sig_registry/registry/models.py

```diff
@@ -141,3 +150,99 @@ def get_hex_display(self):
 
     def get_bytes_display(self):
         return force_text(self.bytes4_signature)
+
+
+class EventSignature(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True, db_index=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    text_signature = models.TextField(unique=True,
+                                        validators=[
+                                            MinLengthValidator(3), MaxLengthValidator(4096)
+                                        ])
+
+    bytes_signature = models.BinaryField(max_length=32,
+                                            unique=False,
```

Comment:
Seems like we would want this field to be unique since.  Duplicates would either be actual duplicates which we don't want, or a hash collision which we assume is improbable enough to not worry about.